src/plusSubscription.ts

```diff
@@ -0,0 +1,50 @@
+import { updateSubscriptionFlags } from './common';
+import createOrGetConnection from './db';
+import {
+  UserSubscriptionStatus,
+  User,
+  type UserSubscriptionFlags,
+} from './entity';
+
+export const updateStoreKitUserSubscription = async ({
+  userId,
+  data,
+  status,
+}: {
+  userId: string;
+  data?: Omit<UserSubscriptionFlags, 'status'>;
+  status?: UserSubscriptionStatus;
+}) => {
+  if (!data) {
+    return;
+  }
+
+  const clone = structuredClone(data);
+
+  // Remove appAccountToken from data, to make sure we don't accidentally update it
+  if (clone.appAccountToken) {
+    delete clone.appAccountToken;
+  }
+
+  const con = await createOrGetConnection();
+
+  // Convert all keys in data to null when status is expired
+  const subscriptionFlags = Object.fromEntries(
+    Object.entries(clone).map(([key, value]) => [
+      key,
+      status === UserSubscriptionStatus.Expired ? null : value,
+    ]),
+  );
```

Comment:
This seems a bit magical, and error prone if something is added without knowing it will be removed, can we specifically set each key to null?

Reply:
Got to enjoy a bit of magic from time to time. I just didn't really like having to do the ternary check for each item like it's done in the paddle version of this.

Reply:
I would rather do it that way because at least its typed then, or since this is sub flags specifically, maybe just set new object with `{ status }` if that is the only thing that needs to be left after sub expires?

Reply:
Yeah, we could just set the status field and remove all other fields

Reply:
Refactored